app/lib/provider_discovery.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+class ProviderDiscovery < OEmbed::ProviderDiscovery
+  USER_AGENT = "#{HTTP::Request::USER_AGENT} (Mastodon/#{Mastodon::VERSION}; +http://#{Rails.configuration.x.local_domain}/)"
```

Comment:
It would be great if we could have some sort of `HTTP` factory because headers and timeout settings have to be set from quite a few places, but I'm not sure how to do that idiomatically. `HttpFactory` doesn't sound like Ruby...